sklearn/metrics/_ranking.py

```diff
@@ -283,14 +284,14 @@ def det_curve(y_true, y_score, pos_label=None, sample_weight=None):
     >>> thresholds
     array([0.35, 0.4 , 0.8 ])
     """
-    if len(np.unique(y_true)) != 2:
-        raise ValueError("Only one class present in y_true. Detection error "
-                         "tradeoff curve is not defined in that case.")
-
     fps, tps, thresholds = _binary_clf_curve(
         y_true, y_score, pos_label=pos_label, sample_weight=sample_weight
     )
 
+    if len(np.unique(y_true)) != 2:
+        raise ValueError("Only one class present in y_true. Detection error "
+                         "tradeoff curve is not defined in that case.")
+
```

Comment:
Why this change? I would imagine it's better to check for it before computing the the `_binary_clf_curve`?

Reply:
As it was, if you pass a multiclass `y_true`, it would rise "Only one class present..." ValueError.

With the reorder, a multiclass ValueError is raised by `_binary_clf_curve`.

Reply:
this needs to be in whats_new since it's changing a behavior.

Reply:
@adrinjalali are you sure?

When passing a multiclass `y_true`, before and now a ValueError is raised; that does not changed. The only change is the error message:
- Before: "Only one class present...", which was wrong.
- Now: "multiclass format is not supported"